File: Students/route.py
from flask import Blueprint
from Students.controller import *

students_bp = Blueprint('students_bp',__name__)

students_bp.route("/getAll",methods=['GET']) (get_all_students)

students_bp.route("/get",methods=['GET']) (get_specific_student)

students_bp.route("/post",methods=['POST']) (insert_student)

students_bp.route("/update",methods=['PATCH']) (update_student)

students_bp.route("/delete",methods=['DELETE']) (del_student)

students_bp.route("/login",methods=['POST']) (login)

File: Students/route.js
const express = require('express')
const studentRouter = express.Router()
const { readAllStudents, readSpecificStudent,signingStudent,updateStudent,deleteStudent,loginStudent, insertMultipleStudents } = require('./controller')
const { validateMiddleware,validateArrayofObjectsMiddleware } = require('./validation')
const { verifyToken } = require('../Utilities/auth')
const { createErrorResponse } = require('../Utilities/errorHandler')

studentRouter.get('/getAll',verifyToken,readAllStudents)
studentRouter.get('/get',verifyToken, validateMiddleware,readSpecificStudent)
studentRouter.post('/insert',verifyToken,validateArrayofObjectsMiddleware,insertMultipleStudents)
studentRouter.post('/signin',validateMiddleware,signingStudent)
studentRouter.put('/update',verifyToken, validateMiddleware,updateStudent)
studentRouter.delete('/delete',verifyToken, validateMiddleware,deleteStudent)
studentRouter.post('/login',validateMiddleware,loginStudent)

studentRouter.use(createErrorResponse)


module.exports = studentRouter
